Scoop/View/EditProfileViewController.swift

```diff
@@ -0,0 +1,256 @@
+            make.height.equalTo(textFieldHeight)
+        }
+        
+        pronounsTextField.textColor = UIColor.black
+        pronounsTextField.borderStyle = .roundedRect
+        stackView.addArrangedSubview(pronounsTextField)
+        
+        pronounsTextField.snp.makeConstraints { make in
+            make.leading.trailing.equalToSuperview()
+            make.height.equalTo(textFieldHeight)
+        }
+        
+        hometownTextField.textColor = UIColor.black
+        hometownTextField.borderStyle = .roundedRect
+        stackView.addArrangedSubview(hometownTextField)
+        
+        hometownTextField.snp.makeConstraints { make in
+            make.leading.trailing.equalToSuperview()
+            make.height.equalTo(textFieldHeight)
+        }
+        
+        classTextField.textColor = UIColor.black
+        classTextField.borderStyle = .roundedRect
+        stackView.addArrangedSubview(classTextField)
+        
+        classTextField.snp.makeConstraints { make in
+            make.leading.trailing.equalToSuperview()
+            make.height.equalTo(textFieldHeight)
+        }
+    }
+    
+    // MARK: - Helper Functions
+    
+    private func updateUser() {
+        let name = nameTextField.text?.split(separator: " ")
+        let firstName = String(name?[0] ?? "")
+        let lastName = String(name?[1...].joined(separator: " ") ?? "")
+        let grade = classTextField.text
+        let pronouns = pronounsTextField.text
+        
+        hometown = hometownTextField.text
+        
+        var userAnswers: [UserAnswer] = []
+                
+        user?.prompts.forEach { prompt in
+            if prompt.questionName == "Hometown" {
+                userAnswers.append(UserAnswer(id: prompt.id, answer: hometown ?? ""))
+            } else if prompt.questionName == "Talkative" {
+                userAnswers.append(UserAnswer(id: prompt.id, answer: String(talkative ?? 0)))
+            } else if prompt.questionName == "Music" {
+                userAnswers.append(UserAnswer(id: prompt.id, answer: String(music ?? 0)))
+            } else if prompt.questionName == "Snack" {
+                userAnswers.append(UserAnswer(id: prompt.id, answer: snack ?? ""))
+            } else if prompt.questionName == "Song" {
+                userAnswers.append(UserAnswer(id: prompt.id, answer: song ?? ""))
+            } else if prompt.questionName == "Stop" {
+                userAnswers.append(UserAnswer(id: prompt.id, answer: stop ?? ""))
+            }
+        }
+        
+        NetworkManager.shared.updateAuthenticatedUser(netid: user?.netid ?? "", first_name: firstName, last_name: lastName, grade: grade ?? "", phone_number: user?.phoneNumber ?? "", pronouns: pronouns ?? "", prof_pic: user?.profilePicUrl ?? "", prompts: userAnswers) { result in
+            switch result {
+            case .success(let user):
+                NetworkManager.shared.currentUser = user
+                self.delegate?.updateUserProfile()
```

Comment:
Strong reference here. Please do `{ [weak self] result in` and then `guard let self = self else { return }`.